packages/wonder-blocks-tooltip/components/tooltip-anchor.test.js

```diff
@@ -0,0 +1,560 @@
+                    expect(result).not.toBeNull();
+                    done();
+                }, 0);
+            };
+
+            arrange(actAndAssert);
+        });
+    });
+
+    test("receives keyboard focus, is active", (done) => {
+        // Arrange
+        const arrange = (actAndAssert) => {
+            const getFakeTooltipPortalMounter = (active) =>
+                (((
+                    <div id="anchor">{active ? "true" : "false"}</div>
+                ): any): React.Element<typeof TooltipPortalMounter>);
+            const nodes = (
+                <div>
+                    <TooltipAnchor anchorRef={actAndAssert}>
+                        {getFakeTooltipPortalMounter}
+                    </TooltipAnchor>
+                </div>
+            );
+            mount(nodes);
+        };
+
+        const actAndAssert = (ref) => {
+            if (!ref) {
+                return;
+            }
+
+            // Act
+            // Let's fake a focusin (this is the event that the anchor gets
+            // whether focused directly or a child is focused). We have to
+            // fake directly because there's no real browser here handling
+            // focus and real events.
+            const event = new FocusEvent("focusin");
+            ref.dispatchEvent(event);
+
+            // Need a timeout here so that the event handling can occur.
+            setTimeout(() => {
+                if (!ref) {
+                    return;
+                }
+
+                const result = ref.innerHTML;
+
+                // Assert
+                expect(result).toBe("true");
+                done();
+            }, 0);
+        };
+
+        arrange(actAndAssert);
+    });
+
+    describe("loses keyboard focus", () => {
+        test("active is set to false", (done) => {
+            // Arrange
+            const arrange = (actAndAssert) => {
+                const getFakeTooltipPortalMounter = (active) =>
+                    (((
+                        <div id="anchor">{active ? "true" : "false"}</div>
+                    ): any): React.Element<typeof TooltipPortalMounter>);
+                const nodes = (
+                    <div>
+                        <TooltipAnchor anchorRef={actAndAssert}>
+                            {getFakeTooltipPortalMounter}
+                        </TooltipAnchor>
+                    </div>
+                );
+                mount(nodes);
+            };
+
+            const actAndAssert = (ref) => {
+                const localRef = ref;
+                if (!localRef) {
+                    return;
+                }
+
+                // Act
+                const focusin = () =>
+                    doEvent(localRef, new FocusEvent("focusin"), focusout);
+                const focusout = () =>
+                    doEvent(localRef, new FocusEvent("focusout"), assert);
+                // Focus starts the event chain defined above.
+                focusin();
```

Comment:
This is kind of hard to read.  Maybe `doEvent` could be rewritten to return a Promise that resolves when the timeout expires.  Then we could use `async/await` to write this code in linear fashion.